php-classes/Emergence/Connectors/Job.php

```diff
@@ -160,40 +143,120 @@ public function logRecordDelta(ActiveRecord $Record, $options = array())
 
         $logEntry['message'] = call_user_func($messageRenderer, $logEntry);
 
-        return $this->log[] = $logEntry;
+        return $this->log(
+            $logEntry['level'],
+            $logEntry['message'],
+            [
+                'changes' => $changes,
+                'record' => $Record
+            ]
+        );
     }
 
     public function logInvalidRecord(\ActiveRecord $Record)
     {
-        return $this->log(array(
-            'message' => 'Invalid '.get_class($Record).' record: '.$Record->getTitle()
-            ,'validationErrors' => $Record->validationErrors
-        ), LogLevel::WARNING);
+        return $this->log(
+            LogLevel::WARNING,
+            'Invalid '.get_class($Record).' record: '.$Record->getTitle(),
+            [
+                'validationErrors' => $Record->validationErrors
+            ]
+        );
     }
 
     public function logException(\Exception $e)
     {
-        return $this->log(array(
-            'message' => get_class($e).': '.$e->getMessage()
-            ,'exception' => $e
-        ), LogLevel::ERROR);
+        return $this->log(
+            LogLevel::ERROR,
+            'Exception:: '.get_class($e).': '.$e->getMessage(),
+            [
+                'exception' => $e
+            ]
+        );
     }
 
     public function getLogPath()
     {
         return $this->isPhantom ? null : \Site::$rootPath.'/site-data/connector-jobs/'.$this->ID.'.json';
     }
 
-    public function writeLog()
+    public function writeLog($logEntry, $compress = false)
     {
         $logPath = $this->getLogPath();
-        $logDirectory = dirname($logPath);
 
+        if (!$logPath) { // record is phantom
+            return;
+        }
+
+        $logDirectory = dirname($logPath);
         if (!is_dir($logDirectory)) {
             mkdir($logDirectory, 0777, true);
         }
 
-        file_put_contents($logPath, json_encode($this->log));
-        exec("bzip2 $logPath");
+        file_put_contents($logPath, json_encode($logEntry), FILE_APPEND | LOCK_EX);
+        if ($compress === true) {
+            exec("bzip2 $logPath");
+        }
+    }
+
+    public function log($level, $message, array $context = [])
+    {
+        $entry = [
+            'message' => $message,
+            'context' => $context,
+            'level' => $level
+        ];
+
+        $this->logEntries[] = $entry;
+        $this->writeLog($entry);
+    }
+
+    /**
+    *  Logger Interface Methods
+    */
+
+    /**
+    * @param string $message
+    * @param array $context
+    * @return null
+    */
+    public function emergency($message, array $context = array())
```

Comment:
@nbey there's a Trait in the Psr library that gives us all these, use that instead.